src/main/webapp/js/ideas/ideas.controller.js

```diff
@@ -1,139 +1,108 @@
-                    }//for
-                }//if
-            });
-
-            return promiseResponse;
-        };
-
-        vm.selectByCategory =function (tag) {
-          tagsFactory.getIdeasByTag(tag).then(function (ideas) {
-            vm.ideas = ideas;
-
-            //$scope.updateGeoObjects(ideas);
-            $scope.geoObjects = mapGeoService.generateGeoObjects(ideas);
-        })
-        };
-
-        ideasFactory.getPage(vm.pageRatingRequest).then(function (ideas) {
-            vm.ideasVisible = ideas;
-
-            //$scope.updateGeoObjects(ideas);
-            $scope.geoObjects = mapGeoService.generateGeoObjects(ideas);
-        });
-
-
-        //load most popular ideas
-        ideasFactory.getPage(vm.pageNewRequest).then(function (ideas) {
-            vm.newIdeas = ideas;
-        });
-/*
-        ideasFactory.getIdeas().then(function (ideas) {
-            vm.ideas = ideas;
-
-            //sort ideas by date
-            vm.ideas.sort(
-                function(a, b){
-                    var keyA = a.rating;
-                    var keyB = b.rating;
-
-                    if(keyA<keyB) return -1;
-                    if(keyA>keyB) return 1;
-                    return 0;
-                }
-            );
-
-            //$scope.updateGeoObjects(ideas);
-            $scope.geoObjects = mapGeoService.generateGeoObjects(ideas);
-
-            //loading array for infinity list
-            if(vm.ideas){
-                vm.ideasVisible = vm.ideas.slice(0, 5);
-            }//if
-        });*/
-
-        $scope.$on('ideas-update', function() {
-            ideasFactory.getIdeas().then(function (ideas) {
-                vm.ideas = ideas;
-
-            //$scope.updateGeoObjects(ideas);
-            $scope.geoObjects = mapGeoService.generateGeoObjects(ideas);
-            });
-        });
-
-        vm.details = function(idea){
-            var ideaDetail = {
-                id:idea.id
-            };
+	'use strict';
+	angular
+		.module('app.controllers')
+		.controller('ideasCtrl', ideasCtrl);
+
+	ideasCtrl.$inject = ['$scope', 'ideasService', 'ideasFactory', 'tagsFactory', 'mapGeoService'];
+
+	function ideasCtrl($scope, ideasService, ideasFactory, tagsFactory, mapGeoService) {
+
+		var vm = this;
+		$scope.geoObjects = null;
+		$scope.criteria = null;
+
+		vm.paramsForPopular = {
+			page: 0,
+			size: 5,
+			sort: 'rating,desc'
+		};
+		vm.paramsForLatest = {
+			page: 0,
+			size: 5,
+			sort: 'creationTime,desc'
+		};
+
+		vm.popular = [];
+		vm.latest = [];
+		vm.tag = null;
+
+		this.loadPageForPopular = function(){
+			vm.paramsForPopular.page++;
+			var promiseResponse = ideasService.getPage(vm.paramsForPopular, vm.tag);
+			promiseResponse.then(function (ideas) {
+				if (ideas) {
+					vm.popular = vm.popular.concat(ideas);
+				}
+			$scope.geoObjects = mapGeoService.generateGeoObjects(vm.popular);
+			});
+			return promiseResponse;
+		};
+
+		this.loadPageForLatest = function(){
```

Comment:
Methods this.loadPageForPopular, this.loadPageForLatest seems to be very similar. May be, common logic can be moved to new method, that allows params (pageNumberRef, ideasRef)?

And, may be, method vm.selectByCategory can be refactored using this new method.